packages/client/clientSchema.graphql

```diff
@@ -105,6 +105,7 @@ extend type EstimateStage {
 }
 
 extend type Discussion {
+  isCreatingNewTask: Boolean
```

Comment:
I think we can get away with removing this if we use https://github.com/ParabolInc/parabol/blob/8c68bf771a350b31c7948e6bd11c5c110e30c977/packages/client/utils/isViewerTyping.js to determine if a task is being edited.

the only difference is `isViewerTyping` will also return true if someone is editing a task, not just creating one (which is a feature, not a bug IMO :wink: )

PS feel free to turn that file into a .ts if you like. some older files are still .js and as we touch them we convert them to typescript

Reply:
@mattkrick that'd be awesome. Any ideas on how can we observe the state of `isViewerTypeing` so we can force React to rerender when it changes? 

Using it that way
```
          <AddTaskButton dataCy={`${dataCy}-add`} onClick={addTask} disabled={isViewerTyping()} />
```
isn't enough unfortunately as when the user is actually adding a new task, the component responsible for showing the `AddTaskButton` isn't rerendering. I googled a bit and I haven't found any obvious answer. I will search more.

Reply:
@mattkrick I've created a new branch with mutation observer implementation: https://github.com/ParabolInc/parabol/compare/feature/4889/task-creation-design...feature/4889/task-creation-design-mutation-observer?expand=1 
Regarding focus handling, I simply reset the state in the `onBlur` comment input handler. Please let me know what do you think 🙏

Reply:
if that works let's go with that!

my cheap & easy alternative would be:
```ts
document.addEventListener('focus', function() {
  setCanCreateTask(!isViewerTyping())
}, true);
```

you could put that in a useEffect & get the same observability without having to monitor the whole DOM tree.

Reply:
@mattkrick ah, thanks for the tip!

I think the cleanest solution is to do the following then
```
  useEffect(() => {
    const focusListener = () => {
      setCanCreateTask(!isViewerTyping())
    }

    document.addEventListener('blur', focusListener, true)
    document.addEventListener('focus', focusListener, true)
    return () => {
      document.removeEventListener('blur', focusListener, true)
      document.removeEventListener('focus', focusListener, true)
    }
  }, [])
```

I'm not sure about the UX because the user will have to click outside of the comment editor to be able to create a task but maybe it's not a big deal.